syntax = "proto3";

import "alice/gamma/sdk/api/api.proto";
import  "google/protobuf/struct.proto";

package api;
option go_package = "a.yandex-team.ru/alice/gamma/sdk/api";

service Sdk {
    rpc Match(MatchRequest) returns (MatchResponse);
}

message MatchRequest {
    message Pattern {
        string pattern = 1;
        string name = 2;
    }

    string input = 1;
    repeated Pattern patterns = 2;
    repeated Entity entities = 3;
}

message MatchResponse {
    message Match {
        string name = 1;
        map<string, Values> variables = 2;
    }

    message Values {
        repeated bytes values = 1;
        repeated google.protobuf.Value protoValues = 2;
    }

    repeated Match matches = 1;
}
